cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/ConditionToConditionAdder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.datapipeline;
+
+import co.cask.cdap.api.customaction.CustomAction;
+import co.cask.cdap.api.workflow.Condition;
+import co.cask.cdap.api.workflow.WorkflowConditionConfigurer;
+
+/**
+ * Implementation of {@link WorkflowProgramAdder} which adds node to the branch of the Condition whose parent is
+ * another Condition.
+ * @param <T> type of the parent
+ * @param <C> type of the current configurer
+ */
+public class ConditionToConditionAdder<T extends WorkflowProgramAdder,
```

Comment:
No need for this to be typed parameter. I started with the configurer as a parent but later realize that we need to send the entire object so changed it back to WorkflowProgramAdder. fixed this now.